src/components/BaseQuestionnaireResponseForm/FormFooter.tsx

```diff
@@ -29,8 +37,27 @@ export function FormFooter(props: Props) {
         className,
         style,
         submitDisabled: initialSubmitDisabled,
+        saveDraftButtonTitle,
+        onSaveDraft,
     } = props;
 
+    const baseQRFPropsContext = useContext(BaseQuestionnaireResponseFormPropsContext);
+    const qrfDataContext = baseQRFPropsContext?.formData.context;
+
+    const formContext = useFormContext();
+
+    const formValues = formContext.getValues();
+    const rootContext = qrfDataContext ? calcInitialContext(qrfDataContext, formValues) : undefined;
```

Comment:
This calculations should happen only once, when handleSaveDraft is clicked.
Now it recalculates for each render.